.github/workflows/ci.yml

```
@@ -12,9 +12,18 @@ jobs:
       fail-fast: false
       matrix:
         runs-on: [ubuntu-latest, windows-latest]
-    name: ${{ matrix.runs-on }}
+        sdk: ["6.0.301", "6.0.302", "6.0.303", "6.0.400"]
+    name: ${{ matrix.runs-on }}-${{ matrix.sdk }}
     runs-on: ${{ matrix.runs-on }}
     steps:
+    - name: Setup .NET Core 3.1 SDK # should always be installed for 3.1 runtime
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 3.1.422
+    - name: Setup .NET Core SDK
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: ${{ matrix.sdk }}
```

```suggestion
    - name: Setup .NET Core SDK
      uses: actions/setup-dotnet@v2
      with:
        dotnet-version:
        - 3.1.422
        - ${{ matrix.sdk }}
```
@sungam3r does this work?

---

Probably, let's try